cfgmgr/buffer_check_headroom_mellanox.lua

```diff
@@ -122,26 +140,50 @@ end
 table.insert(debuginfo, 'debug:other overhead:' .. accumulative_size)
 for pg_key, profile in pairs(all_pgs) do
     local current_profile_size
+    local current_profile_xon
+    local current_profile_xoff
+    local buffer_profile_table_name = 'BUFFER_PROFILE_TABLE:'
     if profile ~= input_profile_name then
-        local referenced_profile_size = redis.call('HGET', 'BUFFER_PROFILE_TABLE:' .. profile, 'size')
+        local referenced_profile_size = redis.call('HGET', buffer_profile_table_name .. profile, 'size')
         if not referenced_profile_size then
-            referenced_profile_size = redis.call('HGET', '_BUFFER_PROFILE_TABLE:' .. profile, 'size')
+            buffer_profile_table_name = '_BUFFER_PROFILE_TABLE:'
+            referenced_profile_size = redis.call('HGET', buffer_profile_table_name .. profile, 'size')
             table.insert(debuginfo, 'debug:pending profile: ' .. profile)
         end
         current_profile_size = tonumber(referenced_profile_size)
+        current_profile_xon = tonumber(redis.call('HGET', buffer_profile_table_name .. profile, 'xon'))
+        current_profile_xoff = tonumber(redis.call('HGET', buffer_profile_table_name .. profile, 'xoff'))
     else
         current_profile_size = input_profile_size
+        current_profile_xon = input_profile_xon
+        current_profile_xoff = input_profile_xoff
     end
     if current_profile_size == 0 then
         current_profile_size = lossy_pg_size
     end
     accumulative_size = accumulative_size + current_profile_size * get_number_of_pgs(pg_key)
-    table.insert(debuginfo, 'debug:' .. pg_key .. ':' .. profile .. ':' .. current_profile_size .. ':' .. get_number_of_pgs(pg_key) .. ':accu:' .. accumulative_size)
+
+    if is_shp_enabled and current_profile_xon and current_profile_xoff then
+        if current_profile_size < current_profile_xon + current_profile_xoff then
+            accumulative_shared_headroom = accumulative_shared_headroom + (current_profile_xon + current_profile_xoff - current_profile_size) * get_number_of_pgs(pg_key)
```

Comment:
Is this calculated only for active ports?

Reply:
Yes